src/uct/ib/rc/verbs/rc_verbs_common.h

```diff
@@ -162,7 +163,16 @@ uct_rc_verbs_iface_poll_rx_common(uct_rc_iface_t *iface)
     UCT_IB_IFACE_VERBS_FOREACH_RXWQE(&iface->super, i, hdr, wc, num_wcs) {
         uct_rc_verbs_iface_handle_am(iface, &wc[i], hdr, wc[i].byte_len);
     }
-    iface->rx.srq.available += num_wcs;
+    iface->rx.srq.rx_req -= num_wcs;
+    /* increase the limit of pre-posted RX requests if "DDoS" detected */
+    if ((num_wcs == iface->super.config.rx_max_poll) &&
```

Comment:
@shamisp Why do you think it will be better? I meant, what is the difference or superiority of the "watermark event" over simple counter?